Added some parts that have not yet been internationalized

fernandozw opened the pull request:
## **User description**
<!--
Selenium IDE is moving to an electron app!!!.
For the time being the team will support both the extension and the app, until the app reaches complete feature parity with the extension.

If you want to submit a PR to the electron app, please submit to master.
To submit a PR to the extension submit to the branch v3
-->

**Thanks for contributing to the Selenium IDE!**
**A PR well described will help maintainers to quickly review and merge it**

Before submitting your PR, please check our [contributing](https://github.com/SeleniumHQ/selenium-ide/blob/master/CONTRIBUTING.md) guidelines.
Avoid large PRs, help reviewers by making them as simple and short as possible.


<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Describe your changes in detail -->

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [contributing](https://github.com/SeleniumHQ/selenium-ide/blob/master/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
<!--- Provide a general summary of your changes in the Title above -->


___

## **Type**
enhancement, documentation


___

## **Description**
- Refactored `CommandLocatorField` component for improved code structure and readability.
- Added new translations for menu items and test core descriptions in `I18N.ts`.
- Enhanced pluralization function to support language-specific rules in `testEditor.ts`.
- Updated dynamic language references in `CommandEditor` and `projectView.ts` for better internationalization support.



___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>LocatorField.tsx</strong><dd><code>Refactor and Enhance CommandLocatorField Component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/browser/windows/ProjectEditor/tabs/Tests/CommandFields/LocatorField.tsx

<li>Refactored the <code>CommandLocatorField</code> component for better readability <br>and maintainability.<br> <li> Updated the state management and event handling within the component.<br> <li> Improved the handling of language-specific labels for UI elements.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1817/files#diff-5acd27841b961b6906da383147a06e31890afcd316d9b594126f97d802afb36c">+1121/-1119</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TestCommandEditor.tsx</strong><dd><code>Integrate Dynamic Language References in CommandEditor</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/browser/windows/ProjectEditor/tabs/Tests/TestCommandEditor.tsx

<li>Integrated dynamic language map references for field names and notes <br>in the <code>CommandEditor</code> component.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1817/files#diff-07c688d33f7d459853b0094e4779718c25ea976d15f251f413d6afbbf0430e70">+6/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>testEditor.ts</strong><dd><code>Enhance Pluralization Function for Internationalization</code>&nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/main/session/controllers/Menu/menus/testEditor.ts

<li>Modified the <code>pluralize</code> function to append 's' based on language <br>settings, enhancing internationalization support.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1817/files#diff-ae35034ad7927ce8818e8bb0a2ecc800cf8a194689f1eb1d796894961890e19f">+6/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>projectView.ts</strong><dd><code>Update Label Retrieval for Menu Item</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/main/session/controllers/Menu/menus/projectView.ts

<li>Updated label retrieval for 'Refresh Playback Window' to support <br>internationalization.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1817/files#diff-9f97ac7e97586e91d5d000f095a574f98b188fc44e2274dd93fbb2ba6130d83d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>I18N.ts</strong><dd><code>Update Translations for Menu Items and Test Core</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
packages/selenium-ide/src/browser/enums/I18N.ts

<li>Added new menu item translations for 'Refresh Playback Window' in both <br>English and Chinese.<br> <li> Introduced new keys for 'Window Handle Name' and its description in <br>the test core section for both languages.<br>


</details>
    

  </td>
  <td><a href="https://github.com/SeleniumHQ/selenium-ide/pull/1817/files#diff-6e6ccd4ae9150222279d1e4dc41464fe4a42b3dbd2444b7aefe4452d4ab828bd">+8/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

fernandozw committed:
Added some parts that have not yet been internationalized

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/SeleniumHQ/selenium-ide?pullRequest=1817) <br/>All committers have signed the CLA.

fernandozw commented:
@toddtarsi In this PR, I have added some parts that have not yet been internationalized.

fernandozw commented:
<img width="983" alt="test11" src="https://github.com/SeleniumHQ/selenium-ide/assets/57780562/93cb26e2-b826-4510-9ce9-52ff99d2978a">

toddtarsi reviewed (commented):
@fernandozw - This LGTM! 🚀 Thank you for the awesome contributions!

For maintenance purposes, I'd like to later source the english commandMap strings from the side-model package here (but we can do later, language support comes first, we can consolidate command maps later.)

https://github.com/SeleniumHQ/selenium-ide/blob/trunk/packages/side-model/src/Commands.ts

Would you also be able to increment packages/selenium-ide? That way we can put out a new binary beta x.9 with your changes?

fernandozw commented:
@toddtarsi I may not add new features for the time being, I will wait for your good news first, haha

toddtarsi commented:
@fernandozw - Absolutely, you are doing great. I actually meant I can do that later for my own cleanup. It's perfect that you're focusing on internationalization. I can't thank you enough for the great work. Let's merge this and I'll increment the ide to cut a release later today

fernandozw committed:
Fixed bug and added restart driver function

fernandozw commented:
@toddtarsi I have a new commit for this pr:
1. Fixed a serious bug. Since startDriver monitors session.driver.stopped, startDriver will always be called as long as session.driver.stopped. However, when this.stopped = false in startProcess, it will cause the driver type to be switched. Unable to switch, it kept prompting that the address is occupied, so I commented this.stopped = false, so that after processing, the driver can switch between electron and chrome at will.
2. As I said, since startDriver listens to session.driver.stopped, after the driver terminates abnormally (such as killing the driver's process through the console), startDriver will not automatically restart, causing the browser to fail to start during playback, so I The restart driver function has been added to System to solve the startup problem after the driver terminates abnormally.
3. In addition, I also processed the default value of Selected Playback Browser when Use Bidi is No. The given default value is electron-version.
4. Also when we switched Use Bidi from No to Yes for the first time, the value of Selected Playback Browser was empty because we did not select the driver at all. I also processed it and gave the default value to electron-version.
5. At this point, Bidi settings will no longer be an Experimental/Non working function. Except for the inability to record through the browser (I want to know your follow-up plans here), other functions can be used normally.

<img width="1055" alt="start" src="https://github.com/SeleniumHQ/selenium-ide/assets/57780562/fc7827cf-ed9f-4448-8029-0c06f3f4edd4">

<img width="1049" alt="index" src="https://github.com/SeleniumHQ/selenium-ide/assets/57780562/a481172d-1fc9-4a71-9080-419d2f4f76cb">

<img width="1310" alt="restart" src="https://github.com/SeleniumHQ/selenium-ide/assets/57780562/23b91276-d392-4c54-b23b-c9686d508a17">

toddtarsi commented on packages/selenium-ide/src/browser/windows/ProjectEditor/tabs/Project/DriverSelect.tsx:
```diff
@@ -1,125 +1,151 @@
+                    }
+                }
+                >
+                    {languageMap.systemConfig.restartDriver}
```
This is so smart!

toddtarsi commented:
Okay, CI is running now. If it passes. we'll merge and increment. You are amazing @fernandozw

toddtarsi commented:
@fernandozw - CI passed, but it looks like you have to sign the Contributor License Agreement thing. Can you sign the CLA using the link in the checks section?

fernandozw committed:
Modify the local git username and email and delete the logs I added

fernandozw commented:
> @fernandozw - CI passed, but it looks like you have to sign the Contributor License Agreement thing. Can you sign the CLA using the link in the checks section?

@toddtarsi I'm very sorry. The username and email address of my previous local commit were different from those of github, which caused it to fail the CLA check. Now the commit information has been modified and you can merge it. Sorry, it was my carelessness.

toddtarsi commented:
@fernandozw - You are 🥇 kicking ass, and this CLA used to bite me all of the time.
